Comparison Engine/Comparison Engine/Child Forms/UserContribution.cs

```diff
@@ -5,32 +5,126 @@
 using System.Data;
 using System.Drawing;
 using System.Text;
+using System.Text.RegularExpressions;
 using System.Windows.Forms;
 
 namespace Comparison_Engine.Child_Forms
 
 {
     public partial class UserContribution : Form
     {
-        public UserContribution(Drink drink, Bar bar)
+        private DrinkManager drinkmanager = DrinkManager.Instance;
+        private BarManager barmanager = BarManager.Instance;
+        public UserContribution()
         {
+            Dictionary<int, float> barChoices = new Dictionary<int, float>();
             InitializeComponent();
+            PopulateComboBoxBar();
+            PopulateComboBoxDrink();
+            comboBoxBar.SelectedIndexChanged += new EventHandler(ComboBoxBarSelectionChange);
+        }
+
+        private void ComboBoxBarSelectionChange(object sender, EventArgs e)
+        {
+            var selectedBar = (Bar)comboBoxBar.SelectedItem;
+            List<Drink> drinklist = new List<Drink>();
+            foreach (var drink in selectedBar.availableDrinks)
+            {
+                drinklist.Add(drinkmanager.GetDrinkByID(drink.Key));
+            }
+            comboBoxDrink.DataSource = new BindingSource(drinklist, null);
         }
 
-        
         private void buttonAddDrink_Click(object sender, System.EventArgs e)
         {
+            if (textBoxDrink.Text == "" || comboBoxBar2.SelectedItem == null )
+            {
+                return;
+            }
+
+            Bar bar = ((Bar)comboBoxBar2.SelectedItem);
+            float price = (float)Convert.ToDouble(textBoxPrice2.Text);
+            string drinkName = textBoxDrink.Text;
+            if (Regex.IsMatch(textBoxPrice2.Text, @"(^[1-9]\d*(.\d{1,2})?$)|(^0(\.\d{1,2})?$)"))
+            {
+               
+                if (drinkmanager.GetDrinkByName(drinkName) == null)
+                {
+
+
+                    drinkmanager.AddDrink(drinkName);
+                    
+                }
+                
+                Drink drink = drinkmanager.GetDrinkByName(drinkName);
+                if (barmanager.GetBarByID(bar.barID).availableDrinks.ContainsKey(drink.drinkID))
+                {
+                    return;
+                }
+
+                barmanager.GetBarByID(bar.barID).AddDrink(drink.drinkID, price);
+                drink.AddBar(bar.barID, price);
+
+            }
+            PopulateComboBoxBar();
+
         }
         private void buttonAddBar_Click(object sender, System.EventArgs e)
         {
             string BarName = textBoxBarName.Text;
             string BarAdress = textBoxBarAdress.Text;
-            //AddBar(BarName, BarAdress);
+            if (BarName == "" || BarAdress == "")
```

Comment:
It would probably be a good idea to check if these values are not null just in case.